protobuf-converter/src/main/java/io/confluent/connect/protobuf/ProtobufData.java

```diff
@@ -0,0 +1,932 @@
+      case FIXED32:
+      case INT64:
+      case UINT64:
+      case SINT64:
+      case FIXED64:
+      case SFIXED64: {
+        builder = SchemaBuilder.int64();
+        break;
+      }
+
+      case FLOAT: {
+        builder = SchemaBuilder.float32();
+        break;
+      }
+
+      case DOUBLE: {
+        builder = SchemaBuilder.float64();
+        break;
+      }
+
+      case BOOL: {
+        builder = SchemaBuilder.bool();
+        break;
+      }
+
+      case STRING:
+        builder = SchemaBuilder.string();
+        break;
+
+      case BYTES:
+        builder = SchemaBuilder.bytes();
+        break;
+
+      case ENUM:
+        builder = SchemaBuilder.int32();
+        EnumDescriptor enumDescriptor = descriptor.getEnumType();
+        builder.name(enumDescriptor.getName());
+        builder.parameter(PROTOBUF_TYPE_ENUM, enumDescriptor.getName());
+        for (EnumValueDescriptor enumValueDesc : enumDescriptor.getValues()) {
+          String enumSymbol = enumValueDesc.getName();
+          String enumTag = String.valueOf(enumValueDesc.getNumber());
+          builder.parameter(PROTOBUF_TYPE_ENUM_PREFIX + enumSymbol, enumTag);
+        }
+        break;
+
+      case MESSAGE: {
+        if (isTimestampDescriptor(descriptor)) {
+          builder = Timestamp.builder();
+          break;
+        }
+
+        builder = toConnectSchema(descriptor.getMessageType(), null);
+        break;
+      }
+
+      default:
+        throw new DataException("Unknown Connect schema type: " + descriptor.getType());
+    }
+
+    if (descriptor.isRepeated() && builder.type() != Schema.Type.MAP) {
+      Schema schema = builder.optional().build();
+      builder = SchemaBuilder.array(schema);
+    }
+
+    builder.optional();
+    builder.parameter(PROTOBUF_TYPE_TAG, String.valueOf(descriptor.getNumber()));
+    return builder.build();
+  }
+
+  private static MessageDefinition timestampDefinition() {
+    MessageDefinition.Builder timestampType = MessageDefinition.newBuilder(
+        GOOGLE_PROTOBUF_TIMESTAMP_NAME);
+    timestampType.addField("optional", "int64", "seconds", 1, null);
+    timestampType.addField("optional", "int32", "nanos", 2, null);
+    return timestampType.build();
+  }
+
+  private static boolean isTimestampDescriptor(FieldDescriptor descriptor) {
+    String name = descriptor.getMessageType().getFullName();
+    return GOOGLE_PROTOBUF_TIMESTAMP_FULL_NAME.equals(name);
+  }
+
+  private static boolean isMapDescriptor(
+      Descriptor descriptor,
+      List<FieldDescriptor> fieldDescriptors
+  ) {
+    return descriptor.getName().endsWith(MAP_ENTRY_SUFFIX)
+        && fieldDescriptors.size() == 2
+        && fieldDescriptors.get(0).getName().equals(KEY_FIELD)
+        && fieldDescriptors.get(1).getName().equals(VALUE_FIELD);
+  }
+
+  private String getNameOrDefault(String name) {
+    return name != null && !name.isEmpty()
+           ? name
+           : DEFAULT_SCHEMA_NAME + (++defaultSchemaNameIndex);
```

Comment:
Question: can this be accessed concurrently ? If so, we should switch to an atomic type for defaultSchemeNameIndex.

Reply:
No it's not accessed concurrently.